### PR TITLE
runner/run-nvim: init

### PR DESCRIPTION
**Pull request by diniamo**
Stuff

### CODE REVIEW

**Comment by NotAShelf** [comment]
virus

**Comment by diniamo** [comment]
dam you got me

File: modules/plugins/runner/run-nvim/config.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        (mkSetLuaLznBinding "n" mappings.runCommand ''
+          function()
+            local input = vim.fn.input("Run command: ")
+            if input ~= "" then require("run").run(input, false) end
+          end
+        '')
+      ];
+    };
+
+    vim.binds.whichKey.register."<leader>r" = mkDefault "+Run";
```
Comment (NotAShelf) [inline modules/plugins/runner/run-nvim/config.nix:34]:
```suggestion
		vim = {
    	lazy.plugins.run-nvim = {
      	package = "run-nvim";
      	setupModule = "run";
      	inherit (cfg) setupOpts;

      	cmd = "Run";

      	keys = [
        	(mkSetLznBinding "n" mappings.run "<cmd>Run<CR>")
        	(mkSetLznBinding "n" mappings.runOverride "<cmd>Run!<CR>")
        	(mkSetLuaLznBinding "n" mappings.runCommand ''
          	function()
            	local input = vim.fn.input("Run command: ")
            	if input ~= "" then require("run").run(input, false) end
          	end
        	'')
      	];
    	};
    };

    binds.whichKey.register."<leader>r" = mkDefault "+Run";
   };
```

Lets not do this again please.

File: modules/plugins/runner/run-nvim/run-nvim.nix [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+in {
+  options = {
+    vim.runner.run-nvim = {
+      enable = mkEnableOption "run.nvim";
+      setupOpts = mkPluginSetupOption "run.nvim" {};
+
+      mappings = {
+        run = mkMappingOption "Run cached" "<leader>ri";
+        runOverride = mkMappingOption "Run and override" "<leader>ro";
+        runCommand = mkMappingOption "Run prompt" "<leader>rc";
```
[the remark this answers is not in the document]
Comment (diniamo) [reply modules/plugins/runner/run-nvim/run-nvim.nix:14]: What do you mean?

**Review by NotAShelf — APPROVED** [review]

**Comment by diniamo** [comment]
Keep in mind that the keymaps still conflict with the Rust ones, and I still haven't come up with a better set.

File: modules/plugins/languages/rust.nix [hunk 3]
```diff
@@ -145,13 +145,13 @@ in {
               on_attach = function(client, bufnr)
                 default_on_attach(client, bufnr)
                 local opts = { noremap=true, silent=true, buffer = bufnr }
-                vim.keymap.set("n", "<leader>rr", ":RustLsp runnables<CR>", opts)
-                vim.keymap.set("n", "<leader>rp", ":RustLsp parentModule<CR>", opts)
-                vim.keymap.set("n", "<leader>rm", ":RustLsp expandMacro<CR>", opts)
-                vim.keymap.set("n", "<leader>rc", ":RustLsp openCargo", opts)
-                vim.keymap.set("n", "<leader>rg", ":RustLsp crateGraph x11", opts)
+                vim.keymap.set("n", "<localleader>rr", ":RustLsp runnables<CR>", opts)
```
Comment (horriblename) [inline modules/plugins/languages/rust.nix:148]: add an option to set <localleader>

also pls document breaking change

**Review by horriblename — CHANGES_REQUESTED** [review]

File: modules/plugins/languages/rust.nix (hunk 3 above)
Comment (diniamo) [reply modules/plugins/languages/rust.nix:148]: We should actually remove vim.leaderKey as well, and encourage users to use vim.globals instead.

File: modules/extra/deprecations.nix [hunk 4]
```diff
@@ -59,5 +59,10 @@ in {
       With Trouble having so many different modes, and breaking changes
       upstream, it no longer makes sense, nor works, to toggle only Trouble.
     '')
+
```
Comment (NotAShelf) [inline modules/extra/deprecations.nix:62]:
```suggestion
# 2024-11-30
```

It's good to note the date of deprecation in order to allow pinpointing the date of the commit.

**Pull request merged**